src/melee/ft/chara/ftMewtwo/ftmewtwo.h

```diff
@@ -9,28 +9,123 @@
 
 #include "melee/ft/fighter.h"
 
-typedef struct _ftMewTwoAttributes {
-    f32 x0;
-    s32 x4;
-    s32 x8;
-    s32 xC;
-    s32 x10;
-    u8 x14_filler[0x2C - 0x14];
-    s32 x2C;
-    u8 x30_filler[0x48 - 0x30];
-    s32 x48;
-    s32 x4C;
-    s32 x50;
-    f32 x54;
-    u8 x58_filler[0x70 - 0x58];
-    s32 x70;
-    s32 x74;
-    s32 x78;
-    s32 x7C;
-    f32 x80;
-    f32 x84;
-} ftMewTwoAttributes; 
+typedef enum ftMewtwoAction {
+
+    AS_FTCOMMON = 340,
+    AS_MEWTWO_SPECIALN_START,
+    AS_MEWTWO_SPECIALN_LOOP,
+    AS_MEWTWO_SPECIALN_FULL,
+    AS_MEWTWO_SPECIALN_CANCEL,
+    AS_MEWTWO_SPECIALN_END,
+    AS_MEWTWO_SPECIALAIRN_START,
+    AS_MEWTWO_SPECIALAIRN_LOOP,
+    AS_MEWTWO_SPECIALAIRN_FULL,
+    AS_MEWTWO_SPECIALAIRN_CANCEL,
+    AS_MEWTWO_SPECIALAIRN_END,
+    AS_MEWTWO_SPECIALS,
+    AS_MEWTWO_SPECIALAIRS,
+    AS_MEWTWO_SPECIALHI_START,
+    AS_MEWTWO_SPECIALHI,
+    AS_MEWTWO_SPECIALHI_LOST,
+    AS_MEWTWO_SPECIALAIRHI_START,
+    AS_MEWTWO_SPECIALAIRHI,
+    AS_MEWTWO_SPECIALAIRHI_LOST,
+    AS_MEWTWO_SPECIALLW,
+    AS_MEWTWO_SPECIALAIRLW,
+
+} ftMewtwoAction;
+
+// Flags used by Mewtwo in Action State Change //
+
+// SpecialHi/SpecialAirHi //
+
+#define FTMEWTWO_SPECIALHI_COLL_FLAG FIGHTER_GFX_PRESERVE | FIGHTER_HIT_NOUPDATE | FIGHTER_MATANIM_NOUPDATE | FIGHTER_COLANIM_NOUPDATE | FIGHTER_CMD_UPDATE | FIGHTER_ITEMVIS_NOUPDATE | FIGHTER_SKIP_UNK_0x2222 | FIGHTER_MODELPART_VIS_NOUPDATE | FIGHTER_MODEL_FLAG_NOUPDATE | FIGHTER_UNK_0x2227
+
+typedef struct _ftMewtwoAttributes {
+
+    // NEUTRAL SPECIAL - SHADOW BALL (SpecialN/SpecialAirN) // 
+
+    f32 x0_MEWTWO_SHADOWBALL_CHARGE_CYCLES; // Number of frames each iteration cycle lasts?
```

Comment:
Uppercase names are typically reserved for things like enums. Don't use it for struct members.